src/elements/content-sidebar/Sidebar.js

```diff
@@ -91,14 +96,63 @@ class Sidebar extends React.Component<Props, State> {
 
         this.state = {
             isDirty: this.getLocationState('open') || false,
+            isDocgenTemplate: false,
         };
 
         this.setForcedByLocation();
     }
 
+    navigateToDefaultTab() {
+        const { history, location } = this.props;
+        if (location.pathname === `/${SIDEBAR_DOCGEN}`) {
+            history.push('/');
+        }
+    }
+
+    fetchMetadataSuccessCallback = ({ editors }: { editors: Array<MetadataEditor> }) => {
+        const { history } = this.props;
+        if (editors && editors.length > 0) {
+            const template = editors[0]?.template;
+            const scopeIsGlobal = template.scope === 'global';
+            const templateKeyIsBoxDocgen = template.templateKey === 'boxDocGen';
+            const isDocgenTemplate = scopeIsGlobal && templateKeyIsBoxDocgen;
+
+            if (isDocgenTemplate) {
+                this.setState({ ...this.state, isDocgenTemplate });
+                // navigate to docgen tab by default
+                history.push(`/${SIDEBAR_DOCGEN}`);
+            }
+        } else {
+            this.navigateToDefaultTab();
+        }
+    };
+
+    fetchMetadata(): void {
+        const { api, file }: Props = this.props;
+
+        if (file && DOCGEN_FILE_TYPES.includes(file.extension)) {
+            api.getMetadataAPI(false).getMetadata(
+                file,
+                this.fetchMetadataSuccessCallback,
+                this.navigateToDefaultTab,
+                this.props.metadataSidebarProps.isFeatureEnabled,
+                { refreshCache: true },
+            );
+        } else {
+            this.navigateToDefaultTab();
+        }
+    }
+
+    componentDidMount() {
+        // if docgen feature is enabled, load metadata to check whether file is a docgen template
+        if (this.props.features?.docgen?.enabled) {
+            this.fetchMetadata();
+        }
+    }
```

Comment:
i'm not familiar with how the Doc Gen feature but should this logic be moved to the DocGenSidebar component?

Reply:
no because this logic determines whether to enable the docgen sidebar in the first place, we have a separate ticket investigating whether this logic can be returned via monolith call that preview makes in order to avoid making extra metadata call - in that case most of this logic could be removed